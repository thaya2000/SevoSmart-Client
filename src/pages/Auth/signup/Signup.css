.signup-container {
  display:flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 25rem;
  height: 38rem;
  background-color: rgb(255, 255, 255);
  position: relative;
  font-weight: 600;
  font-size: 2rem;
  margin-bottom: 100px;
  margin-top: 100px;
  border:2px solid rgb(193, 189, 189);
  backdrop-filter: blur(5px);
  margin-left: 38%;

  
}

.register-input {
  display: block;
  width: 300px; /* Adjust width to accommodate the icon */
  height: 2.2rem;
  margin-top: 25px;
  margin-bottom: 20px;
  /* border-radius: 1rem; */
  padding: 1rem;
  font-weight: 300;
  font-size: 1rem;
  background-color: rgb(255, 255, 255);
  border: 2px solid rgba(4, 4, 4, 0.2);
}


.create-account-text{
  margin-top: -100px;
  margin-bottom: 10px;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  color: rgb(122, 124, 124);
  text-decoration-skip-ink: none; /* Ensure underline covers entire text */
  padding-bottom: 5px; /* Increase space beneath the text */
  border-bottom: 3px solid rgb(83, 85, 86);
}

.submit-button-register{
  top:80%;
  width: 300px; /* Adjust width as needed */
  height: 35px; /* Adjust height as needed */
  color: rgb(251, 250, 250); /* Set text color */
  border: 2px solid rgb(149, 143, 143); /* Remove border */
  display: flex; /* Use flexbox */
  justify-content: center; /* Center horizontally */
  align-items: center; /* Center vertically */
  cursor: pointer; /* Add cursor pointer for interaction */
  position: absolute; /* Position absolute */
  font-size: 18px;
  margin-top: 40px;
  background-color:  rgb(9, 33, 81);
  left: 50%; /* Center horizontally */
  transform: translate(-50%, -50%); /* Centering trick */
  transition: background-color 0.3s ease; /* Smooth transition */
}

/* Hover effect */
.submit-button-register:hover {
  background-color: rgb(130, 131, 133); /* Change background color on hover */
}

.input-container{
  justify-content: center;
  align-items: center;
  border:rgb(255, 255, 255) 2px solid;
  padding: 20px;
  
}
.login-line{
 
  font-size: 14px;
 margin-left:50px;
}
.login-line {
  display: flex;
  align-items: center;
}

.login-line span {
  margin-right: 10px; 
}

.link-style{
  color: blue; 
  text-decoration: underline;

}