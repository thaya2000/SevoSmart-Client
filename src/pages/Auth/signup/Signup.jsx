import { useState } from "react";
import "./Signup.css";
import { userAuth } from "../../../context/authContext";
import { useNavigate, useLocation } from "react-router-dom";
import axios from "axios";
import toast from "react-hot-toast";

export default function Signup() {
  const [firstname, setFirstName] = useState("user_f");
  const [lastname, setLastName] = useState("user_l");
  const [email, setEmail] = useState("user@gmail.com");
  const [role, setRole] = useState("USER");
  const [password, setPassword] = useState("user1234");

  // hook
  const [auth, setAuth] = userAuth();
  const navigate = useNavigate();
  const location = useLocation();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      const { data } = await axios.post(
        `http://localhost:8080/api/v1/auth/register`,
        {
          firstname,
          lastname,
          email,
          role,
          password,
        }
      );
      console.log(data);
      if (data?.message) {
        console.log(data.message);
        toast.error(data.message);
      } else {
        localStorage.setItem("auth", JSON.stringify(data));
        setAuth({ ...auth, token: data.token, user: data.user });
        navigate(location.state || "/");
        toast.success("Registration successful");
      }
    } catch (err) {
      console.log(err);
      toast.error("Registration failed. Try again.");
    }
  };

  return (
    <div className="flex relative items-center justify-center w-100v h-100v">
      <div className="signup-container">
        <div className="create-account-text">SignUp</div>
        <form onSubmit={handleSubmit} className="flex flex-col">
          <input
            type="text"
            className="register-input"
            placeholder="Enter your first name"
            value={firstname}
            onChange={(e) => setFirstName(e.target.value)}
            autoFocus
          />
          <input
            type="text"
            className="register-input"
            placeholder="Enter your last name"
            value={lastname}
            onChange={(e) => setLastName(e.target.value)}
          />
          <input
            type="text"
            className="register-input"
            placeholder="Enter your email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
          />
          <input
            type="text"
            className="register-input"
            placeholder="Enter your role"
            value={role}
            onChange={(e) => setRole(e.target.value)}
          />
          <input
            type="text"
            className="register-input"
            placeholder="Enter your password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
          />
          <div
            className="submit-button-register "
            onClick={handleSubmit}
          >
            SignUp
          </div>
        </form>
      </div>
    </div>
  );
}
