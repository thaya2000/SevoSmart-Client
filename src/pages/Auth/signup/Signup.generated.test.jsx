import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import Signup from './Signup';

// Mock CSS and modules
vi.mock("./Signup.css");
vi.mock("../../../context/authContext", () => ({
  userAuth: vi.fn(() => [{}, vi.fn()]) // Mock userAuth with a state and a setter function
}));
vi.mock("react-router-dom", () => ({
  useNavigate: vi.fn(),
  useLocation: vi.fn(() => ({ state: {} }))
}));
vi.mock("axios");
vi.mock("react-hot-toast", () => ({
  toast: vi.fn()
}));
vi.mock("../../../util/constants", () => ({
  DEFAULT_EMAIL: "default@example.com",
  DEFAULT_PASSWORD: "defaultPassword",
  DEFAULT_ROLE: "user",
  DEFAULT_FIRSTNAME: "John",
  DEFAULT_LASTNAME: "Doe"
}));

describe('<Signup>', () => {
  it('should render component', () => {
    const { asFragment } = render(<Signup />);
    expect(asFragment()).toMatchSnapshot();
  });
});
