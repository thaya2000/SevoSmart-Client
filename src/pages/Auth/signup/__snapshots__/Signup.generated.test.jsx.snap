// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Signup> > should render component 1`] = `
<DocumentFragment>
  <div
    class="flex relative items-center justify-center w-100v h-100v"
  >
    <div
      class="signup-container"
    >
      <div
        class="create-account-text"
      >
        SignUp
      </div>
      <form
        class="flex flex-col"
      >
        <input
          class="register-input"
          placeholder="Enter your first name"
          type="text"
          value="John"
        />
        <input
          class="register-input"
          placeholder="Enter your last name"
          type="text"
          value="Doe"
        />
        <input
          class="register-input"
          placeholder="Enter your email"
          type="text"
          value="default@example.com"
        />
        <input
          class="register-input"
          placeholder="Enter your role"
          type="text"
          value="user"
        />
        <input
          class="register-input"
          placeholder="Enter your password"
          type="text"
          value="defaultPassword"
        />
        <div
          class="submit-button-register "
        >
          SignUp
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
